packages/cache/src/S3CacheStorage.ts

```diff
@@ -0,0 +1,172 @@
+import path from "path";
+import { PassThrough, pipeline, Transform, TransformCallback } from "stream";
+import tarFs from "tar-fs";
+import { Logger } from "backfill-logger";
+import { stat } from "fs-extra";
+import { GetObjectCommand, S3Client } from "@aws-sdk/client-s3";
+import { Upload } from "@aws-sdk/lib-storage";
+import { S3CacheStorageOptions } from "backfill-config";
+import { CacheStorage } from "./CacheStorage";
+
+/*
+ * Timeout stream, will emit an error event if the
+ * input has not started providing data after a given time after
+ * its creation.
+ */
+class TimeoutStream extends Transform {
```

Comment:
Can you move these helper streams to a shared file instead of copying them from `AzureBlobCacheStorage.ts`?

Reply:
Done